### PR TITLE
Fix some broken links of owasp's top 10 project.

### PR DESCRIPTION
**Pull request by songzy12**
Some of the current url links for OWASP's top 10 project are broken.

This pull request fixes them to be the current avaliable corresponding pdf pages.

### CODE REVIEW

**Comment by digininja** [comment]
Can we do something more up to date than 2007?

**Comment by songzy12** [comment]
Do you mean a more up-to-date reference for the file inclusion vulnarability?

I searched for "file inclusion site:owasp.org", the most proper pages seem to be:
- https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/11.1-Testing_for_Local_File_Inclusion
- https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/11.2-Testing_for_Remote_File_Inclusion

If we also want the constraint of owasp's "top 10 project", the following one is somehow relevant but not exactly matched:
- https://owasp.org/www-pdf-archive/OWASP_Top_10-2017_%28en%29.pdf.pdf#page10

**Comment by digininja** [comment]
For file inclusion, yes, go with the two you mention, they are the up-to-date OWASP guides. If you could also include a working link to the Top Ten that had file inclusion in, that would be good, just for historic reference.

I'd drop the links into PDFs, stick to standard web pages if you can, they are more accessible.

For the weak ID, can you find something similarly modern?

**Comment by songzy12** [comment]
For weak session id, I did a quick search and found this on owasp:
- https://owasp.org/www-community/attacks/Session_Prediction

Do you think it is suitable?

**Comment by digininja** [comment]
Have they migrated that page to the new site yet?

Maybe include that page but also add this one as well as it is on the new
system and so will probably be maintained for longer

https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html

**Comment by songzy12** [comment]
Oops I did not notice the site migration announcement.

Maybe then we can use the following links for weak_id:
- https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/06-Session_Management_Testing/01-Testing_for_Session_Management_Schema
- https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html

**Comment by digininja** [comment]
They look good to me.

If you fancy looking through, and updating, any of the other help or
reference links while you are doing these, it would be really appreciated.

**Comment by songzy12** [comment]
Sure. Thanks for your quick response!

I will update the pull request with the above links later. ^_^

**Comment by digininja** [comment]
No rush, thanks for the work.

**Comment by songzy12** [comment]
For multiple references in one `help.php` file, I searched in the current code base and found this:

https://github.com/ethicalhack3r/DVWA/blob/30741c892d47030dbdda658e6bb4d240a287d3dd/vulnerabilities/csp/help/help.php#L49-L51

Not sure whether this is the right routine to follow.

**Comment by digininja** [comment]
Without downloading and testing it, it looks OK. Does it render correctly?

**Comment by songzy12** [comment]
It now looks like:

**Comment by songzy12** [comment]
The current pull request is ready for review.

With a quick search, I found some other broken owasp links where the challenges I have not practiced. 
Maybe I could send some other pull requests in the future.

**Comment by digininja** [comment]
Think you might have done something wrong as the link is showing not the
link text.

**Comment by digininja** [comment]
We rarely turn away pull requests!

**Comment by songzy12** [comment]
ah, that screenshot is about the new code and the above reference is about some exisitng code.

For the existing code, the corresponding screenshot is like:

**Comment by digininja** [comment]
That is correct, we show the link text not the link. As long as yours matches that it will be good.

**Comment by songzy12** [comment]
Update: link text is added for the changed links.

**Comment by digininja** [comment]
Let me know when you are done making changes and I'll go through them all
at once when I've got five minutes.

**Comment by songzy12** [comment]
Sure. I think it is done and ready for review for this PR now. Thanks!

**Comment by digininja** [comment]
OK, will look later.

If I've not replied by the end of the week, give me a reminder.

File: vulnerabilities/fi/help/help.php [hunk 1]
```diff
@@ -58,6 +58,7 @@
 
 	<br />
 
-	<p>Reference: <?php echo dvwaExternalLinkUrlGet( 'https://www.owasp.org/index.php/Top_10_2007-A3' ); ?></p>
+	<p>Reference: <?php echo dvwaExternalLinkUrlGet( 'https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/11.1-Testing_for_Local_File_Inclusion', 'WSTG - Local File Inclusion' ); ?></p>
```
Comment (digininja) [inline vulnerabilities/fi/help/help.php:61]: Looks good and works

File: vulnerabilities/fi/file3.php [hunk 2]
```diff
@@ -20,10 +20,11 @@
 		[<em><a href=\"?page=include.php\">back</a></em>]
 	</div>
 
-	<h2>More info</h2>
+	<h2>More Information</h2>
 	<ul>
-		<li>" . dvwaExternalLinkUrlGet( 'https://en.wikipedia.org/wiki/Remote_File_Inclusion' ) . "</li>
-		<li>" . dvwaExternalLinkUrlGet( 'https://www.owasp.org/index.php/Top_10_2007-A3' ) . "</li>
+	<li>" . dvwaExternalLinkUrlGet( 'https://en.wikipedia.org/wiki/Remote_File_Inclusion', 'Wikipedia - File inclusion vulnerability' ) . "</li>
```
Comment (digininja) [inline vulnerabilities/fi/file3.php:25]: Can you add this link to the help.php as well. May as well give all references in all places.

File: vulnerabilities/fi/file1.php [hunk 3]
```diff
@@ -11,10 +11,11 @@
 		[<em><a href=\"?page=include.php\">back</a></em>]
 	</div>
 
-	<h2>More info</h2>
+	<h2>More Information</h2>
```
Comment (digininja) [inline vulnerabilities/fi/file1.php:14]: All good

File: vulnerabilities/fi/file2.php [hunk 4]
```diff
@@ -9,10 +9,11 @@
 		\"<em>I needed a password eight characters long so I picked Snow White and the Seven Dwarves.</em>\" ~ Nick Helm<br /><br />
 		[<em><a href=\"?page=include.php\">back</a></em>]	</div>
 
-	<h2>More info</h2>
+	<h2>More Information</h2>
```
Comment (digininja) [inline vulnerabilities/fi/file2.php:12]: All good

File: vulnerabilities/fi/include.php [hunk 5]
```diff
@@ -22,8 +22,9 @@
 
 	<h2>More Information</h2>
 	<ul>
-		<li>" . dvwaExternalLinkUrlGet( 'https://en.wikipedia.org/wiki/Remote_File_Inclusion' ) . "</li>
-		<li>" . dvwaExternalLinkUrlGet( 'https://www.owasp.org/index.php/Top_10_2007-A3' ) . "</li>
+	<li>" . dvwaExternalLinkUrlGet( 'https://en.wikipedia.org/wiki/Remote_File_Inclusion', 'Wikipedia - File inclusion vulnerability' ) . "</li>
```
Comment (digininja) [inline vulnerabilities/fi/include.php:25]: All good

File: vulnerabilities/weak_id/help/help.php [hunk 6]
```diff
@@ -35,5 +35,6 @@
 
 	</div>
 
-	<p>Reference: <?php echo dvwaExternalLinkUrlGet( 'https://www.owasp.org/index.php/Top_10_2013-A2-Broken_Authentication_and_Session_Management' ); ?></p>
+	<p>Reference: <?php echo dvwaExternalLinkUrlGet( 'https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/06-Session_Management_Testing/01-Testing_for_Session_Management_Schema', 'WSTG - Session Management Schema' ); ?></p>
```
Comment (digininja) [inline vulnerabilities/weak_id/help/help.php:38]: Looks good

**Comment by digininja** [comment]
All looks good.

I think add the one extra  FI help that is on the main page to the help page and that should be it.

**Comment by songzy12** [comment]
Ack. Done.

**Pull request merged**